glommio/src/io/buffered_file_stream.rs

```diff
@@ -654,7 +654,12 @@ impl AsyncBufRead for Stdin {
     ) -> Poll<io::Result<&'a [u8]>> {
         match self.source.take() {
             Some(source) => {
-                let res = source.result().unwrap();
+                let res = match source.result() {
+                    Some(res) => res,
+                    None => {
+                        return Poll::Pending;
```

Comment:
I don't know if this is the correct fix, but it does fix both the test case and when using `stdin` in my program to read from it